janitor/functions/factorize_columns.py

```diff
@@ -13,50 +14,49 @@ def factorize_columns(
     **kwargs,
 ) -> pd.DataFrame:
     """
-    Converts labels into numerical data
+    Converts labels into numerical data.
 
     This method will create a new column with the string `_enc` appended
     after the original column's name.
     This can be overriden with the suffix parameter.
 
-    Internally this method uses pandas `factorize` method.
+    Internally, this method uses pandas `factorize` method.
     It takes in an optional suffix and keyword arguments also.
     An empty string as suffix will override the existing column.
 
     This method mutates the original DataFrame.
```

Comment:
Should we make it non mutating? Does it matter? Is it outside d scope of this PR?

Reply:
Agreed, non-mutating is a very good idea. 

I think the way to not mutate the original column is to use a `.assign(col_name=some_value)` instead of `df[col_name] = some_value`.

@thatlittleboy if you'd like to sneak in the change here, that'd be great! Otherwise, we can just raise another issue. :smile:.

Reply:
Just as an aside to be wary of `assign`, as it only supports strings. #1016

Reply:
Oh! That’s right, good catch. Then copying the data frame seems to be the way out, I guess.